lib/connectors/highlevel_client.go

```diff
@@ -0,0 +1,341 @@
+	result, err := c.client.ResumeConnector(req)
+	if err != nil {
+		return result, err
+	}
+
+	if sync {
+		if !tryUntil(
+			func() bool {
+				resp, err := c.GetConnectorStatus(req)
+				return err == nil && resp.Code == 200 && resp.ConnectorStatus["state"] == "RUNNING"
+			},
+			2*time.Minute,
+		) {
+			return result, errors.New("timeout on resuming connector sync")
+		}
+	}
+	return result, nil
+}
+
+//IsUpToDate checks if the given configuration is different from the deployed one.
+//Returns true if they are the same
+func (c *highLevelClient) IsUpToDate(connector string, config map[string]interface{}) (bool, error) {
+	// copy the map to safely interact with it
+	// we are going to need to add connector name to be able to exact match
+	copyConfig := make(map[string]interface{}, len(config))
+	for key, value := range config {
+		copyConfig[key] = value
+	}
+
+	copyConfig["name"] = connector
+
+	configResp, err := c.GetConnectorConfig(ConnectorRequest{Name: connector})
+	if err != nil {
+		return false, err
+	}
+	if configResp.Code == 404 {
+		return false, nil
+	}
+	if configResp.Code >= 400 {
+		return false, errors.New(fmt.Sprintf("status code: %d", configResp.Code))
+	}
+
+	if len(configResp.Config) != len(copyConfig) {
+		return false, nil
+	}
+	for key, value := range configResp.Config {
+		if convertConfigValueToString(copyConfig[key]) != convertConfigValueToString(value) {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+// Because trying to compare the same field on 2 different config may return false negative if one is encoded as a string and not the other
+func convertConfigValueToString(value interface{}) string {
+	switch v := value.(type) {
+	case string:
+		return v
+	case int:
+		return strconv.Itoa(v)
+	default:
+		return ""
+	}
+}
+
+// tryUntil repeats exec until it return true or timeout is reached
+// tryUntil itself return true if `exec` has return true (success), false if timeout (failure)
+func tryUntil(exec func() bool, limit time.Duration) bool {
+	timeLimit := time.After(limit)
+
+	run := true
+	defer func() { run = false }()
+	success := make(chan bool)
+	go func() {
+		for run {
+			if exec() {
+				success <- true
+				return
+			}
+			time.Sleep(1 * time.Second)
+		}
+	}()
+
+	select {
+	case <-timeLimit:
+		return false
+	case <-success:
+		return true
+	}
+}
+
+//DeployConnector checks if the configuration changed before deploying.
+//It does nothing if it is the same
+func (c *highLevelClient) DeployConnector(req CreateConnectorRequest) (err error) {
+	existingConnector, err := c.GetConnector(ConnectorRequest{Name: req.Name})
+	if err != nil {
+		return err
+	}
+
+	if existingConnector.Code != 404 {
+		var upToDate bool
+		upToDate, err = c.IsUpToDate(req.Name, req.Config)
+		if err != nil {
+			return err
+		}
+		// Connector is already up to date, stop there and return ok
+		if upToDate {
+			return nil
+		}
+
+		_, err = c.PauseConnector(ConnectorRequest{Name: req.Name}, true)
```

Comment:
this part (and the resume) should be done only if the connector is actually running, if it's paused, the deployment should not take the responsibility to resume it, only update it

Reply:
usually it is not much of a problem, but yes you are right.

Reply:
i will keep it that way, right now.
Not that it is not a problem, but i think we can actually remove this part.
Would rather do it in another PR